xpdan/data_reduction.py

```diff
@@ -140,14 +148,71 @@ def construct_event_stream(self, header):
 xpd_data_proc = DataReduction()
 ai = AzimuthalIntegrator()
 
+
+def associate_dark(header, events, handler):
+    dark_uid = header.start.get(an_glbl.dark_field_key, None)
+    if dark_uid is None:
+        print("INFO: no dark frame is associated in this header, "
+              "subrraction will not be processed")
+        dark_img = None
+    else:
+        dark_search = {'uid': dark_uid}
+        dark_header = handler.exp_db(**dark_search)[0]
+        dark_img = handler.exp_db.get_images(dark_header,
+                                             handler.image_field)[0]
+    for e in events:
+        yield dark_img
+
```

Comment:
That's fair. Although we might discuss the long term viability of the handler at some point.